src/filter/segfilter.cpp

```diff
@@ -1,6 +1,21 @@
 #include "segfilter.hpp"
 #include "../Exporter/exportimage.hpp"  // Include ExportImage header
 
+SegFilter::SegFilter(const std::vector<std::string>& modelPaths, ThreadLogger* logger)
+        : logger(logger) {
+    for (const auto& path : modelPaths) {
+        IModel* model = new TensorRTModel();
```

Comment:
Also we have to make sure the models DONT fit in the memory as loading a model takes around a minute (which is way too long)